From e8170b1620eb5f1ddc1479d3f9ba0f82cbfde2f8 Mon Sep 17 00:00:00 2001
From: Le Yang <le.yang@amlogic.com>
Date: Tue, 10 Feb 2015 19:26:06 +0800
Subject: [PATCH 5437/5965] PD#103139: amports: fixed the unvalid pcr use to
 sync.

In pcrmaster the decode may output frame before the system_pcr init. so the invalid
pcr may cause frame show like freerun.

Change-Id: Ia2a5e492d3597d2ec24493ca38107c17fa90d685
---
 drivers/amlogic/amports/video.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/amlogic/amports/video.c b/drivers/amlogic/amports/video.c
index c8a8f1a69404..17650531c8a6 100755
--- a/drivers/amlogic/amports/video.c
+++ b/drivers/amlogic/amports/video.c
@@ -2486,6 +2486,10 @@ static inline bool vpts_expire(vframe_t *cur_vf, vframe_t *next_vf)
     u32 systime;
     u32 adjust_pts, org_vpts;
 
+    if (timestamp_pcrscr_enable_state() == 0 && tsync_get_mode() == TSYNC_MODE_PCRMASTER) {
+        return false;
+    }
+
     if(debug_flag & DEBUG_FLAG_TOGGLE_FRAME_PER_VSYNC){
         return true;
     }
-- 
2.19.0

